src/lib.rs

```diff
@@ -62,9 +48,25 @@ pub enum ClangFormatError {
 }
 
 /// Execute clang-format with the given input, using the given style, and collect the output
-fn clang_format_with_style(
+///
+/// # Example
+///
+/// ```
+/// # use clang_format::{clang_format_with_style, ClangFormatStyle};
+/// # fn main() {
+/// let input = r#"
+///     struct Test {
+///
+///     };
+/// "#;
+/// let output = clang_format_with_style(input, ClangFormatStyle::Mozilla);
+/// assert!(output.is_ok());
+/// assert_eq!(output.unwrap(), "\nstruct Test\n{};\n");
+/// # }
+/// ```
+pub fn clang_format_with_style(
```

Comment:
I think if we remove `clang_format`, we should rename this function to `clang_format`.

However, I'm not sure we should remove the `clang_format` function.
As it could just call `clang_format_with_style` with the default style and would make for a nice and easy-to-use API.
Also, if we add the ability to configure more flags later, the `clang_format` function would still be useful, as it would just mean "clang-format with whatever the default is".

Reply:
So maybe instead we should port to OnceCell from std ?

Reply:
No, I think there isn't really a point to the OnceCell anymore.
If you want to set a style, just call `with_style`.

But there should be a function for "I don't care about the style, or other settings, just use whatever the default is".
That would be `clang_format`, which would just use the `Default` style.

Reply:
Removed the OnceCell and made `clang_format` use `Default` style.